pkg/controller/mizar/mizar-service-controller.go

```diff
@@ -264,9 +319,16 @@ func (c *MizarServiceController) processServiceDeletion(key string) error {
 
 	response := GrpcDeleteService(c.grpcHost, msg)
 	code := response.Code
-	if code != CodeType_OK {
-		klog.Errorf("Return Code: %v", code)
-		return errors.New("Service Deletion failed on mizar side")
+	switch code {
+	case CodeType_OK:
+		klog.Infof("Mizar handled service deletion successfully: %s", key)
+	case CodeType_TEMP_ERROR:
+		klog.Infof("Mizar hit temporary error for service deletion for service: %s", key)
+		c.queue.AddRateLimited(eventKey)
```

Comment:
Does it work? The processNextWorkItem() expects the item got from the queue is of type KeyWithEventType. Yet you add an object of type string to the queue here. 

```
func (c *MizarServiceController) processNextWorkItem() bool {
	workItem, quit := c.queue.Get()
	...
       eventKey := workItem.(KeyWithEventType)
```

Seems this issue exist in all the c.queue.AddRateLimited() in this PR.